### PR TITLE
Add test support (setupLink, linkFor, TestLink)

### PR DESCRIPTION
**Pull request by tstormk**

### CODE REVIEW

File: addon/helpers/link.ts [hunk 1]
```diff
@@ -115,10 +117,12 @@ export default class LinkHelper extends Helper {
...
         : named.model
         ? [named.model]
-        : (positional.slice(
+        : positional.length > 0
+        ? (positional.slice(
             1,
             positionalQueryParameters ? -1 : undefined
-          ) as RouteModel[]),
+          ) as RouteModel[])
+        : undefined,
```
Comment (buschtoens) [inline addon/helpers/link.ts]: Why was this necessary? Shouldn't an empty array work?

File: addon-test-support/test-link.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (this.onReplaceWith) {
+      this.onReplaceWith();
+    }
+  }
+
+  private _preventTransitionOut(event: Event) {
+    // Make sure we don't transition out of the testing page
+    event && event.preventDefault();
+  }
```
Comment (buschtoens) [inline addon-test-support/test-link.ts]: Should we rather apply the same safe-guards here, so we don't cancel _any_ event?

File: tests/unit/test-helpers/test-link-test.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    properties.forEach(propertyName => {
+      const propertyKey = propertyName as keyof TestLink;
+
+      assert.notOk(link[propertyKey]);
+
+      // Ignore `Cannot assign to '[property]' because it is a read-only property.`
+      /* eslint-disable-next-line @typescript-eslint/ban-ts-ignore */
+      // @ts-ignore
+      link[propertyKey] = true;
```
[the remark this answers is not in the document]
Comment (tstormk) [reply tests/unit/test-helpers/test-link-test.ts]: No, this is simply a Typescript misunderstanding when using this `keyof`/looping approach, because technically `keyof TestLink` could be any property of `TestLink`, it flags with this error

simply setting the developer-relevant properties (such as `isActive`) won't result in this error

File: addon-test-support/test-link.ts (hunk 2 above)
Comment (buschtoens) [reply addon-test-support/test-link.ts]: Ignore me, I thought this does more like checking for the `target.tagName` etc.

https://github.com/buschtoens/ember-link/blob/b8dce02499506f1d02f4aa76b1fb91107389d6a7/addon/link.ts#L260-L267
